File: code/ch09/9.2.2.1.long_substr_without_repeat_char.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            for (int j = i; j < s.length(); j++) {
                String tmp = s.substring(i, j + 1);
                if (allUnique(tmp))
                    res = Math.max(res, 1 + j - i);
            }
        }
        return res;
    }

    public boolean allUnique(String s) {
        Set<Character> set = new HashSet<>();
        for (int i = 0; i < s.length(); i++) {
            if (set.contains(s.charAt(i)))
                return false;
            else
                set.add(s.charAt(i));
        }
        return true;
    }
}

File: code/ch09/9.2.2.1.long_substr_without_repeat_char.py
class Solution:
    def allUnique(self, s:str)->bool:
        dic = {}
        for i in range(0, len(s), 1):
            if s[i] in dic:
                return False
            else:
                dic[s[i]] = 1

        return True

    def lengthOfLongestSubstring(self, s:str)->int:
        res = 0

        for i in range(0, len(s), 1):
            for j in range(0, len(s), 1):
                if self.allUnique(s[i : j + 1]):
                    res = max(res, j - i + 1)
                else:
                    break

        return res
